File: Bridge/Bridge/ConsoleApplication4/Properties/Shapess/Square.cs
using System;

namespace Shapes
{
    public class Square:Rectangle
    {
        public Square(double side, IColor color) : base(side, side, color)
        {
            if (side <= 0)
            {
                throw new ArgumentException("Side length must be positive.");
            }
        }
    }
}

File: Bridge/Petukhov/src/lnu/edu/ua/shapes/Square.java
package lnu.edu.ua.shapes;

import lnu.edu.ua.colors.Color;

public class Square extends Rectangle {
    public Square(Color color, double side) {
        super(color, side, side);
    }



    public String toString() {
        StringBuilder str = new StringBuilder("Square:\n");
        str.append("side: ").append(this.firstSide).append("\n");
        str.append("perimeter: ").append(getPerimeter()).append("\n");
        str.append("area: ").append(getArea()).append("\n");
        str.append("color: ").append(this.color.getColor()).append("\n");
        return str.toString();
    }
}
